sale_automatic_workflow_advance/models/sale_workflow_job.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2022 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+from odoo.tools.safe_eval import safe_eval
+
+from odoo.addons.sale_automatic_workflow.models.automatic_workflow_job import savepoint
+
+
+class AutomaticWorkflowJob(models.Model):
+    _inherit = "automatic.workflow.job"
+
+    @api.model
+    def run_with_workflow(self, sale_workflow):
+        workflow_domain = [("workflow_process_id", "=", sale_workflow.id)]
+        if sale_workflow.automated_advance_payment_create:
+            self._automated_advance_payment_create(
+                safe_eval(sale_workflow.advance_payment_filter_id.domain)
+                + workflow_domain
+            )
+        return super(AutomaticWorkflowJob, self).run_with_workflow(
+            sale_workflow=sale_workflow
+        )
+
+    @api.model
+    def _automated_advance_payment_create(self, order_filter):
+        sales = self.env["sale.order"].search(order_filter)
+        for sale in sales:
+            with savepoint(self.env.cr):
+                self._do_automated_advance_payment_create(
+                    sale.with_company(sale.company_id), order_filter
+                )
+
+    def _do_automated_advance_payment_create(self, sale, domain_filter):
+        if not self.env["sale.order"].search_count(
+            [("id", "=", sale.id)] + domain_filter
+        ):
+            return "{} {} job bypassed".format(sale.display_name, sale)
+        else:
+            ctx = {
+                "active_ids": sale.ids,
+                "default_order_id": sale.id,
+                "default_amount_total": sale.amount_residual,
+                "default_currency_id": sale.pricelist_id.currency_id.id,
+            }
+            bank_journal = (
```

Comment:
IMO this code is not optimal. You cannot take the first one arbitrarily.
The best method would be to use https://github.com/OCA/bank-payment/tree/15.0/account_payment_mode features to get a proper journl

Reply:
@Cedric-Pigeon 
Odoo standard uses this approach in several occasions, so it is line with that.
We don't want to introduce the account_payment_mode dependency.
What I suggest is to have this in a `_prepare_...` method, that can then be extended to support that or other logic.

Reply:
@dreispt Yes, at least that should be welcome (even if I agree with @Cedric-Pigeon and taking Odoo example is not necessary a good thing - if there is no good reason :smile: )